crypto/crypto_test.cc

```diff
@@ -27,6 +27,25 @@
 #include <gtest/gtest.h>
 #include "test/test_util.h"
 
+static int AWS_LC_ERROR_return(void) {
+  GUARD_PTR(NULL);
+  return 1;
+}
+
+static int AWS_LC_SUCCESS_return(void) {
+  char non_null_ptr[1];
+  GUARD_PTR(non_null_ptr);
+  return 1;
+}
+
+TEST(CryptoTest, SafetyMacro) {
```

Comment:
We've noticed that `GUARD_PTR` also works great for functions that return a pointer and they'll return `NULL` when checking pointers.
ex:
```
// Will return `NULL`.
static void *AWS_LC_ERROR_return(void) {
  GUARD_PTR(NULL);
  return 1;
}
```
Should we consider adding a test for that?